### PR TITLE
Profile.java, ProfileStore.java, edits to other files

### PR DESCRIPTION
**Pull request by jeoratz**

### CODE REVIEW

**Review by SophiaJefferson — APPROVED** [review]
This looks really good! I'm not sure why there is a function to create random profiles but that is the only thing I don't understand

File: src/main/java/codeu/model/data/User.java [hunk 1]
```diff
@@ -60,6 +60,11 @@ public Instant getCreationTime() {
     return creation;
   }
 
+  /**
+   * Resets the password of this User.
+   */
+  public void setPassword(String password) { this.password = password; }
```
Comment (gauravmishra) [inline src/main/java/codeu/model/data/User.java:66]: does this need to be hashed?

**Review by gauravmishra — APPROVED** [review]
good stuff! the model & store looks good...
Now you'd need to add a servlet mapping to web.xml, create a servlet to serve the requests and a jsp to view the profiles...

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        Profile profile = profileStore.getProfile(username);
+        if (profile == null) {
+            // couldn't find profile, redirect to homepage
+            System.out.println("Profile was null: " + username);
+            response.sendRedirect("/index.jsp");
+            return;
+        }
+        request.setAttribute("profileUser", user);
+        request.setAttribute("profile", profile);
```
Comment (gauravmishra) [inline src/main/java/codeu/controller/ProfileServlet.java]: do you need this?

**Review by Faramawy009 — APPROVED** [review]
Skimmed over the code then downloaded the branch and  set up a dev server of and it looks great!

File: src/main/java/codeu/controller/ProfileServlet.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    Profile profile = profileStore.getProfile(username);
+    if (profile == null) {
+      // couldn't find profile, redirect to homepage
+      response.sendRedirect("/index.jsp");
+      return;
+    }
+    if (request.getParameter("update") != null) {
+      String newAbout = request.getParameter("newAbout");
+      profile.setAbout(newAbout);
```
Comment (gauravmishra) [inline src/main/java/codeu/controller/ProfileServlet.java:111]: @sgj1 An example here on how to update the ProfileStore when updating a Profile. Similarly for after password change, you'd need to update the User object in the UserStore. Also see changes in the profileStore

File: src/main/java/codeu/model/store/persistence/PersistentDataStore.java [hunk 4]
```diff
@@ -176,4 +216,22 @@ public void writeThrough(Conversation conversation) {
     conversationEntity.setProperty("creation_time", conversation.getCreationTime().toString());
     datastore.put(conversationEntity);
   }
+
+  /** Write a Profile object to the Datastore service. */
+  public void writeThrough(Profile profile) {
+    // Delete profile in entity if already present
```
Comment (gauravmishra) [inline src/main/java/codeu/model/store/persistence/PersistentDataStore.java:222]: @sgj1 see here for the changes needed in the writeThrough method for User

**Pull request merged**